packages/api/test/user-payment.spec.js

```diff
@@ -154,11 +155,46 @@ describe('userPaymentPut', () => {
       customers.mockCustomers.map(c => c.id).includes(billing.paymentMethodSaves[0].customerId),
       'billing.paymentMethodSaves[0].customerId is in customers.mockCustomers')
   })
+  it('throws an error if no agreement', async function () {
+    const desiredPaymentMethodId = `pm_${randomString()}`
+    const paymentSettings = { paymentMethod: { id: desiredPaymentMethodId }, subscription: { storage: null } }
+    const authorization = createBearerAuthorization(AuthorizationTestContext.use(this).createUserToken())
+    const request = createMockAuthenticatedRequest(createSaveUserPaymentSettingsRequest({ authorization, body: JSON.stringify(paymentSettings) }))
+    const billing = createMockBillingService()
+    const customers = createMockCustomerService()
+    const env = {
+      ...createMockBillingContext(),
+      billing,
+      customers
+    }
+    assert.rejects(
+      async () => await userPaymentPut(request, env),
+      /Missing Terms of Service/
+    )
```

Comment:
Nice! Thanks for testing this stuff. Love a good regression test ensuring there is a grokkable error message 🎉